zebra-state/src/service.rs

```diff
@@ -252,6 +252,87 @@ impl StateService {
     pub fn utxo(&self, outpoint: &transparent::OutPoint) -> Option<transparent::Output> {
         self.mem.utxo(outpoint).or_else(|| self.sled.utxo(outpoint))
     }
+
+    /// Return an iterator over the relevant chain of the block identified by
+    /// `hash_or_height`.
+    #[allow(dead_code)]
+    pub fn chain(&self, hash_or_height: HashOrHeight) -> Iter<'_> {
```

Comment:
But it's still important that we document the precise semantics of the `chain` function.

I think they are:
```suggestion
    /// Return an iterator over the relevant chain of the block identified by
    /// `hash_or_height`. Heights iterate over the best chain. Hashes iterate
    /// over the chain containing the hash, which can be the best chain or a
    /// side chain. 
    #[allow(dead_code)]
    pub fn chain(&self, hash_or_height: HashOrHeight) -> Iter<'_> {
```